### PR TITLE
test/utils: replace Eventually loop with API call

### PR DESCRIPTION
**Pull request by dankenigsberg**
Replace possibly-long and possibly-racey Eventually loops with with the relevant API call (Stop or Start).

/sig code-quality

```release-note
NONE
```

### CODE REVIEW

File: tests/utils.go [hunk 1]
```diff
@@ -303,14 +295,8 @@ func StartVirtualMachine(vm *v1.VirtualMachine) *v1.VirtualMachine {
...
 
-	Eventually(func() error {
-		updatedVM, err := virtClient.VirtualMachine(vm.Namespace).Get(context.Background(), vm.Name, metav1.GetOptions{})
-		Expect(err).ToNot(HaveOccurred())
-		updatedVM.Spec.Running = pointer.P(true)
-		updatedVM.Spec.RunStrategy = nil
-		_, err = virtClient.VirtualMachine(updatedVM.Namespace).Update(context.Background(), updatedVM, metav1.UpdateOptions{})
-		return err
-	}, 300*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+	err := virtClient.VirtualMachine(vm.Namespace).Start(context.Background(), vm.Name, &v1.StartOptions{})
```
[the remark this answers is not in the document]
Comment (EdDev) [reply tests/utils.go:298]: Also, this `vm.*` reference is probably wrong and needs treatment in some follow up PR.

[the remark this answers is not in the document]
Comment (dankenigsberg) [reply tests/utils.go:298]: > Also, this vm.* reference is probably wrong and needs treatment in some follow up PR.

Would you educate me why? `StartVirtualMachine()` attempts to start a `vm`. Why shouldn't it refer to `vm.*`?

[the remark this answers is not in the document]
Comment (EdDev) [reply tests/utils.go:298]: Starting a VM assumes here that the namespace is defined, which is not really assured, especially when using libvmi.

Like with other client-go operations, the namespace should be passed separately.

[the remark this answers is not in the document]
Comment (dankenigsberg) [reply tests/utils.go:298]: > Starting a VM assumes here that the namespace is defined, which is not really assured, especially when using libvmi.

`StartVirtualMachine()` does not create the VM, it assumes that it is already created. As such it must already have a valid namespace that can be used.

[the remark this answers is not in the document]
Comment (EdDev) [reply tests/utils.go:298]: Ok, valid... but you assume it passes an updated VM object.
The operation requires only the namespace and name, passing the whole object is not really needed and I still claim it is not that safe.

**Comment by arnongilboa** [comment]
@dankenigsberg the fix is already covered by #12489

**Pull request closed without merging**